README.md

```diff
@@ -1,3 +1,93 @@
 # FHIR to HPO
 
-This library will facilitate the conversion of FHIR messages (observations) to HPO Terms. This is in the early stages of developments and is being adapted from the core library here https://github.com/monarch-initiative/loinc2hpo/tree/master/loinc2hpo-core.
\ No newline at end of file
+This library converts FHIR STU3 Observations to Human Phenotype Ontology (HPO) Terms when LOINC codes and interpretable values are present.
+
+Using the jar requires local installation of the [monarch-initiative phenol library](https://github.com/monarch-initiative/phenol) that defines the domain around the HPO. 
+
+Once phenol is installed, compile the library and run the tests using `mvn clean install`.
+
+## Using the library
+
+fhir2hpo can be included in your maven project as a dependency:
+
+```
+<dependency>
+	<groupId>org.monarchinitiative</groupId>
+	<artifactId>fhir2hpo</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+</dependency>
+```
+
+Use the ObservationAnalysisService provided by this library to perform conversion. In a Spring Boot application, you can autowire this service and access the associated domain by scanning the fhir2hpo packages.
+
+```
+package org.myapp.pkg;
+
+@SpringBootApplication
+@EntityScan(basePackages = { "org.myapp.pkg", "org.monarchinitiative.fhir2hpo" })
+@ComponentScan({ "org.myapp.pkg", "org.monarchinitiative.fhir2hpo" })
+public class MyApp {
+
+	@Autowired
+	ObservationAnalysisService observationAnalysisService;
+
+	public static void main(String[] args) {
+		SpringApplication.run(MyApp.class, args);
+	}
+}
+```
+
+Then pass STU3 observations to the service to convert:
+
+```
+Observation observation = getObservationFromFhirServer(...);
+List<HpoConversionResult> hpoConversionResult = observationAnalysisService.convert(observation);
+```
+
+## Understanding the HpoConversionResult
+
+The conversion of an observation to an HpoTerm can fail for a variety of reasons. Conversion might also succeed using one method but fail using another. The entity HpoConversionResult is used to encapsulate the full context of the run so the consumer has detailed information and can decide how to handle.
+
+The convert method actually passes back a list of conversion results. This is because in rare cases an Observation might contain multiple Loinc codes, and it is possible that these codes are annotated differently in our system. See the below example:
```

Comment:
I know it's the convention and there really isn't a better way around it but I always find myself cringing a little at "Loinc codes" since the "C" is "Codes". :)

Reply:
Also, should "Loinc" be capitalized?